nbconvert/exporters/tests/test_templateexporter.py

```diff
@@ -114,6 +117,35 @@ def test_relative_template_file(self):
             exporter = self._make_exporter(config=config)
             assert os.path.abspath(exporter.template.filename) == template
             assert os.path.dirname(template) in [ os.path.abspath(d) for d in exporter.template_path ]
+    
+    def test_in_memory_template(self):
+        # Loads in an in memory template using jinja2.DictLoader
+        # creates a class that uses this template with the template_file argument
+        # converts an empty notebook using this mechanism
+        my_loader = DictLoader({'my_template': "{%- extends 'rst.tpl' -%}"})
+        
+        class MyExporter(TemplateExporter):
+            template_file = 'my_template'
+        
+        exporter = MyExporter(extra_loaders=[my_loader])
+        nb = v4.new_notebook()
+        out, resources = exporter.from_notebook_node(nb)
+
+
+    @pytest.mark.xfail(strict=True, raises=TemplateNotFound)
```

Comment:
As a stylistic point, I think it's nicer to use `pytest.raises()`, making it specific to the statement we expect to raise an error, rather than marking the whole test as an expected failure. More info in docs here:

http://doc.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

I'm not going to hold up merging for this, though.